src/v/kafka/server/group.cc

```diff
@@ -1856,6 +1858,13 @@ group::prepare_tx(cluster::prepare_group_tx_request r) {
       raft::replicate_options(raft::consistency_level::quorum_ack));
 
     if (!e) {
+        // Situation: replication has passed but the replicate method returns
```

Comment:
in general it is impossible for replication to return an error while entry has been replicated Why do we need a special case ?

Reply:
I am not sure here.
@rystsov could you help, please?

Need to get more info about replication errors

Reply:
Protective programming, in general I can imagine that a replicate call may time out with the unknown result, even if now it doesn't time out and hang for even we may change it later e.g. to save resources. Also it's safe to assume that any method may an return. Having this logic doesn't hurt so I prefer to keep it.